fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -2451,4 +2451,43 @@ private void checkSplitIndexBuildRange(int minSplit, int maxSplit,
             commit(context);
         }
     }
+
+    @Test
+    public void testMockedLocalityUtil() throws Exception {
+        final FDBDatabaseFactory factory = FDBDatabaseFactory.instance();
+        factory.setLocalityProvider(MockedLocalityUtil.instance());
+        FDBDatabase database = FDBDatabaseFactory.instance().getDatabase();
+
+        try (FDBRecordContext context = database.openContext()) {
+            openSimpleRecordStore(context, TEST_SPLIT_HOOK);
+            ArrayList<byte[]> keys = new ArrayList<>();
+            for (int i = 0; i < 50; i++) {
+                keys.add(recordStore.recordsSubspace().pack(i));
+            }
+
+            MockedLocalityUtil.init(keys, 0);
+            assertEquals(0, getRangeCount(MockedLocalityUtil.instance().getBoundaryKeys(context.ensureActive(), keys.get(0), keys.get(keys.size() - 1))));
+
+            MockedLocalityUtil.init(keys, 1);
+            assertEquals(1, getRangeCount(MockedLocalityUtil.instance().getBoundaryKeys(context.ensureActive(), keys.get(0), keys.get(keys.size() - 1))));
+
+            int rangeCount = new Random().nextInt(keys.size() + 1);
+            MockedLocalityUtil.init(keys, rangeCount);
+            assertEquals(rangeCount, getRangeCount(MockedLocalityUtil.instance().getBoundaryKeys(context.ensureActive(), keys.get(0), keys.get(keys.size() - 1))));
+
+            MockedLocalityUtil.init(keys, keys.size());
+            assertEquals(keys.size(), getRangeCount(MockedLocalityUtil.instance().getBoundaryKeys(context.ensureActive(), keys.get(0), keys.get(keys.size() - 1))));
+
+            commit(context);
+        }
+    }
+
+    private int getRangeCount(@Nonnull CloseableAsyncIterator<byte[]> iterator) {
```

Comment:
I guess you can use Guavva's `Iterators.size(iterator)`

Reply:
Good point. Done.